src/test/java/uk/gov/companieshouse/overseasentitiesapi/model/DtoModelChangeTest.java

```diff
@@ -15,37 +16,75 @@
  * clients of the OE API end-points will deal with older versions of the model structure and ultimately how older
- * versions of the model structure can be loaded into the newly changed DAO and DTO models. Changes will probably
- * need to be made to the Mongo custom 'converters'.
+ * versions of the model structure can be loaded into the newly changed DAO and DTO models. Among other things, a new,
+ * Mongo, custom 'transformer' class may need to be written.
  *
- * Once those things have been considered and addressed, existing tests should pass again. Then, a file with the new JSON
- * model structure should be added to the test resources folder and a new test added to successfully read and process
- * the new, up-to-date DAO/DTO structure.
+ * If a non-breaking change to the model has been made - e.g. a new data field has simply been added to a DTO and DAO -
+ * the 'current' test JSON file needs updating to avoid a null check failure (this essentially future-proofs the test,
+ * which will potentially fail if the datatype of the newly added field ever changes).
  *
- * Note the use of Spring profiles, to ensure that the (default) <code>MongoConfig</code> configuration is not loaded
- * during this test (as the specific in-memory Mongo DB config needs to be picked up).
+ * If a breaking change to the model has been made - e.g. a data field has changed its type from String to a custom POJO -
+ * after a new 'transformer' has been implemented, a file with the new JSON model structure should be added to the test
+ * resources folder and the <code>OLD_JSON_MODEL_FILENAMES</code> and <code>CURRENT_JSON_MODEL_FILENAME</code> constants
+ * updated, in order that the new, up-to-date DAO/DTO structure is fully tested.
+ *
+ * Finally, note the use of Spring profiles, to ensure that the (default) <code>MongoConfig</code> configuration is not
+ * loaded during this test (as the specific in-memory Mongo DB config needs to be picked up).
  */
 @AutoConfigureDataMongo
 @SpringBootTest
 @ActiveProfiles("test")
 class DtoModelChangeTest {
 
+    private static final Logger LOGGER = Logger.getLogger(DtoModelChangeTest.class.getName());
+
+    private static final String[] HISTORICAL_JSON_MODEL_FILENAMES = {
+            "overseas_entity_v_1_0.json",
+            "overseas_entity_v_2_0.json",
+            "overseas_entity_v_3_0.json" };
+
+    /**
+     * If the DTO model changes and it is a 'breaking change', the current filename must be added to the
+     * <code>HISTORICAL_JSON_MODEL_FILENAMES</code> list and a new test JSON file will need to be created and then
+     * referenced here.
+     */
+    private static final String CURRENT_JSON_MODEL_FILENAME = "overseas_entity_v_3_1.json";
+
+    /**
+     * Allows the list to be used as input to a JUnit parameterised test (see below).
+     */
+    private static List<String> getHistoricalJsonModelFilenames() {
```

Comment:
Could just declare a static array list and then below have:

static { 
    arrayList.add("overseas_entity_v_1_0.json")
    ...
}
etc.


}